ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/CommandTopicTest.java

```diff
@@ -0,0 +1,282 @@
+    verify(future).get();
+  }
+
+  @Test
+  public void shouldThrowExceptionIfSendIsNotSuccessfull() throws Exception {
+    // Given:
+    when(future.get())
+        .thenThrow(new ExecutionException(new RuntimeException("Send was unsuccessful!")));
+    expectedException.expect(RuntimeException.class);
+    expectedException.expectMessage("Send was unsuccessful!");
+
+    // When
+    commandTopic.send(commandId1, command1);
+  }
+
+  @Test
+  public void shouldThrowRuntimeExceptionIfSendCausesRunTimeException() throws Exception {
+    // Given:
+    when(future.get()).thenThrow(new ExecutionException(
+        new Exception("Send was unsuccessful because of non RunTime exception!")));
+    expectedException.expect(RuntimeException.class);
+    expectedException.expectMessage(
+        "java.lang.Exception: Send was unsuccessful because of non RunTime exception!");
+
+    // When
+    commandTopic.send(commandId1, command1);
+  }
+
+  @Test
+  public void shouldThrowRuntimeExceptionIfSendThrowsInterruptedException() throws Exception {
+    // Given:
+    when(future.get()).thenThrow(new InterruptedException("InterruptedException"));
+    expectedException.expect(RuntimeException.class);
+    expectedException.expectMessage("InterruptedException");
+
+    // When
+    commandTopic.send(commandId1, command1);
+  }
+
+  @Test
+  public void shouldGetNewCommandsIteratorCorrectly() {
+    // Given:
+    when(commandConsumer.poll(any(Duration.class))).thenReturn(consumerRecords);
+
+    // When:
+    final Iterable<ConsumerRecord<CommandId, Command>> newCommands = commandTopic
+        .getNewCommands(Duration.ofHours(1));
+
+    // Then:
+    assertThat(newCommands, sameInstance(consumerRecords));
+  }
+
+  @Test
+  public void shouldGetRestoreCommandsCorrectly() {
+    // Given:
+    when(commandConsumer.poll(any(Duration.class)))
+        .thenReturn(someConsumerRecords(
+            new ConsumerRecord<>("topic", 0, 0, commandId1, command1),
+            new ConsumerRecord<>("topic", 0, 0, commandId2, command2)))
+        .thenReturn(someConsumerRecords(
+            new ConsumerRecord<>("topic", 0, 0, commandId3, command3)))
+        .thenReturn(new ConsumerRecords<>(Collections.emptyMap()));
+
+    // When:
+    final List<QueuedCommand> queuedCommandList = commandTopic
+        .getRestoreCommands(Duration.ofMillis(1));
+
+    // Then:
+    verify(commandConsumer).seekToBeginning(topicPartitionsCaptor.capture());
+    assertThat(topicPartitionsCaptor.getValue(),
+        equalTo(Collections.singletonList(new TopicPartition(COMMAND_TOPIC_NAME, 0))));
+    assertThat(queuedCommandList, equalTo(ImmutableList.of(
+        new QueuedCommand(commandId1, command1, Optional.empty()),
+        new QueuedCommand(commandId2, command2, Optional.empty()),
+        new QueuedCommand(commandId3, command3, Optional.empty()))));
+  }
+
+
+  @Test
+  public void shouldGetRestoreCommandsCorrectlyWithDuplicateKeys() {
```

Comment:
Question for my own understanding: why would we have duplicate keys in the command topic?